build/Makefile

```diff
@@ -136,90 +111,29 @@ system-keys : ${PDO_ENCLAVE_CODE_SIGN_PEM}
 ${PDO_ENCLAVE_CODE_SIGN_PEM} :
 	openssl genrsa -3 -out ${PDO_ENCLAVE_CODE_SIGN_PEM} 3072
 
-service_indexes = 1 2 3 4 5
-
-ifeq ($(PDO_LEDGER_TYPE),ccf)
-	ESERVICE_KEYS := $(addprefix $(KEYDIR),$(foreach i,$(service_indexes),eservice$(i)_private.pem))
-	ecdsa_curve := secp384r1
-endif
-
-SSERVICE_PEM = $(addprefix $(KEYDIR)/,$(foreach i,$(service_indexes),sservice$(i)_private.pem))
-PSERVICE_PEM = $(addprefix $(KEYDIR)/,$(foreach i,$(service_indexes),pservice$(i)_private.pem))
-
-user_indexes = 1 2 3 4 5 6 7 8 9 10
-USER_PEM = $(addprefix $(KEYDIR)/,$(foreach i,$(user_indexes),user$(i)_private.pem))
-
-%.pem :
-	. $(abspath $(DSTDIR)/bin/activate) && $(KEYGEN) --keyfile $(subst _private,,$*) --format pem --curve $(ecdsa_curve)
+# SERVICES_COUNT is the number of services of each type to create
+# First value is the number of eservices, then pservices, then
+# sservices, 5 of each is the default
+SERVICES_COUNT ?= 5 5 5
+SERVICES_CONF_TEMPLATES = $(addprefix $(SCRIPTDIR)/template/, eservice.toml pservice.toml enclave.toml)
+SERVICES_CONF_TARGET = $(DSTDIR)/opt/pdo/.services_configured
 
-service-keys : $(KEYDIR) $(ESERVICE_KEYS) $(SSERVICE_PEM) $(PSERVICE_PEM)
+$(SERVICES_CONF_TARGET) : $(PYTHON_DIR) $(SERVICE_CONF_TEMPLATES)
+	. $(abspath $(DSTDIR)/bin/activate) && \
+		pdo-configure-services -o $(DSTDIR)/opt/pdo -t $(SCRIPTDIR)/template -c $(SERVICES_COUNT)
+	touch $(SERVICES_CONF_TARGET)
 
-user-keys : $(KEYDIR) $(USER_PEM)
-
-keys : service-keys user-keys
-
-ESERVICE_CONF = $(ETCDIR)/eservice1.toml
-SSERVICE_CONF = $(ETCDIR)/sservice1.toml
-PSERVICE_CONF = $(ETCDIR)/pservice1.toml
-ENCLAVE_CONF = $(ETCDIR)/enclave.toml
-CONTRACT_CONF = $(ETCDIR)/pcontract.toml
-
-GENERATED_CONF_FILES = $(ESERVICE_CONF) $(SSERVICE_CONF) $(PSERVICE_CONF) $(ENCLAVE_CONF) $(CONTRACT_CONF)
-
-ESERVICE_CONF_TEMPLATE = $(SCRIPTDIR)/opt/pdo/etc/template/eservice.toml
+config-services : $(SERVICES_CONF_TARGET)
 
 verified-config : verify-pre-config
 	${MAKE} config
 
-config : $(ETCDIR) $(GENERATED_CONF_FILES) $(ETCDIR)/site.psh
+config : config-services config-client
```

Comment:
Do we expect this make target to be used outside of local testing? I'm wondering if we should rename this target to `local-config` or something that reflects this.

Reply:
i debated whether to rename a lot of these with something like a "test-" prefix. Most are really only relevant for a test configuration anyway. I decided to hold off and do a big Makefile update separately (to capture all the other flow changes we have been discussing).

Reply:
Got it. Do we have an issue where we're tracking all these changes?